### ISSUE TITLE
BUG：GPU calling failure when starting first clusting

### ISSUE THREAD

**Issue by Alchemist-Y** [MouseLand/Kilosort#743]
### Describe the issue:

It is normal during the spike extracting, but when it comes to the firsting clustering part, kilosort will automatically choose the cpu to computer rather than the GPU.

### Reproduce the bug:

```python
GPU calling is falied during the first clustering
```

### Version information:

kilosort: latest version; CUDA toolkit: 11.8; NVIDIA driver: windows server 2022 standard; GUP: NVIDIA RTX A2000 12GB

**Comment by jacobpennington** [comment]
I'm sorry, I'm having a hard time understanding the issue. What do you mean it "automatically chooses the cpu?" How are you determining that? Are you getting an error during sorting? What version of Kilosort4 is this?

**Comment by Alchemist-Y** [comment]
@jacobpennington  Hi，I,m sorry about my stodgy description. I mean, I choose the GPU as for the pytorch in the GUI, but during the clustering part, the GPU doesn't work at all observing in my task manager. And inversely, the CPU is 100% working.

**Comment by jacobpennington** [comment]
The task manager is not an accurate way to check GPU usage for python processes. If you want to check usage while sorting is running, you can use the `nvidia-smi` command in a terminal or powershell.

**Comment by Alchemist-Y** [comment]
@jacobpennington Hi，here I show the detail information about the GPU usage: extracting spike part and clustering part.
![image](https://github.com/user-attachments/assets/79df5cff-3488-4898-bb05-b79f08b78e69)
![image](https://github.com/user-attachments/assets/1465a894-bbb6-45f8-8d49-3430226e19c9)

**Comment by jacobpennington** [comment]
Okay. Can you please clarify if you get an error during sorting, or think it's taking too long, or some other problem? Not all steps of the sorting process use the GPU.

**Comment by Alchemist-Y** [comment]
> Okay. Can you please clarify if you get an error during sorting, or think it's taking too long, or some other problem? Not all steps of the sorting process use the GPU.

Yes, it takes too long to process in the first clustering part. Because in my own computer with GPU-NVIDIA 1650S 4GB, rather than in this win server, it can  solve the entire sorting in about 3h, but in this environment, it takes 2 hours in the firsting clustering part and without any processing.

**Comment by jacobpennington** [comment]
Are you sure your pytorch installation is set up correctly? You said you're using CUDA toolkit version 11.8, but your screenshots show version 12.4.

**Comment by Alchemist-Y** [comment]
@jacobpennington hi，this figure showed above means the highest edition CUDA which my server support, and what I've installed in CUDA 11.8 showed under.
![image](https://github.com/user-attachments/assets/74317c56-1047-46ee-89f6-323c23195a51)

**Comment by jacobpennington** [comment]
Ah, I see. Unfortunately, I'm not sure what else I can try to debug for you if the sorting works fine on a different machine. The next thing I would try is the following, just to make sure nothing got messed up with the installation and there aren't any conflicts with other packages:

1) Restart the machine.
2) Create a new conda environment to use only for Kilosort (it looks like you're using the base environment right now).
3) Follow the steps in the readme again to install Kilosort and pytorch.
4) Retry the sorting.

If you still get the same error after trying that, please upload `kilosort4.log` from the results directory from the new sorting attempt. Screenshots of the Kilosort4 GUI with that recording loaded might also be helpful, if you're using the GUI.

**Comment by Alchemist-Y** [comment]
Hi，jacob，here is my log. Because it takes too long to process the fisrt clustering part, so I interrupt it. 
[kilosort4.log](https://github.com/user-attachments/files/16755922/kilosort4.log)

**Comment by RobertoDF** [comment]
I have exactly the same problem, first clustering extremely slow because no GPU usage. On Task manager I can isolate CUDA usage and it goes down at the "first clustering" step. I am calling kilosrt directly, not via spikeinterface. I reinstalled CUDA toolkit and pytorch but no improvement.

**Comment by RobertoDF** [comment]
Here a video of the issue
https://www.dropbox.com/scl/fi/vaipnzqoxsyrrvbkuy3o8/Untitled.m4v?rlkey=watrlnib7q6a30a3sbzqovpoc&dl=0

**Comment by Alchemist-Y** [comment]
> Here a video of the issue https://www.dropbox.com/scl/fi/vaipnzqoxsyrrvbkuy3o8/Untitled.m4v?rlkey=watrlnib7q6a30a3sbzqovpoc&dl=0

same with yours

**Comment by RobertoDF** [comment]
I could solve this by using a totally fresh environment, not just uninstall>reinstall. However now I get the CUDA out of memory error 😰

**Comment by jacobpennington** [comment]
Still unclear what's causing this since I can't reproduce it, but given that it appears to work for you on some machines and was resolved in at least one case by creating a new environment, I'm going to close the issue and recommend using `environment.yml` to create a new environment if you continue to have this problem. Instructions for that are at the bottom of the readme.
